npm/puppeteer/README.md

```diff
@@ -0,0 +1,368 @@
+# @cypress/puppeteer
+
+Utilize [Puppeteer's browser API](https://pptr.dev/api) within Cypress with a single command.
+
+# Table of Contents
+
+- [Installation](#installation)
+- [Compatibility](#compatibility)
+- [Usage](#usage)
+- [API](#api)
+- [Examples](#examples)
+- [Contributing](#contributing)
+- [Changelog](./CHANGELOG.md)
+
+# Installation
+
+## npm
+
+```sh
+npm install --save-dev @cypress/puppeteer
+```
+
+## yarn
+
+```sh
+yarn add @cypress/puppeteer
+```
+
+## With TypeScript
+
+Add the following in `tsconfig.json`:
+
+```json
+{
+  "compilerOptions": {
+    "types": ["cypress", "@cypress/puppeteer"]
+  }
+}
+```
+
+## Compatibility
+
+`@cypress/puppeteer` requires Cypress version 13.6.0 or greater.
+
+Only Chromium-based browsers (e.g. Chrome, Chromium, Electron) are supported.
+
+## Usage
+
+`@cypress/puppeteer` is set up in your Cypress config and support file, then executed in your spec. See [API](#api) and [Examples](#examples) below for more details.
+
+While the `cy.puppeteer()` command is executed in the browser, the majority of the Puppeteer execution is run in the Node process via your Cypress config. You pass a string message name to `cy.puppeteer()` that indicates which message handler to execute in the Cypress config. This is similar to how [cy.task()](on.cypress.io/task) operates.
+
+In your Cypress config (e.g. `cypress.config.ts`):
+
+```typescript
+import { setup } from '@cypress/puppeteer/plugin'
+
+export default defineConfig({
+  e2e: {
+    setupNodeEvents (on) {
+      setup({
+        on,
+        onMessage: {
+          async switchToTabAndGetContent (browser) {
+            // ...
+          },
+        },
+      })
+    },
+  },
+}
+```
+
+In your support file (e.g. `cypress/support/e2e.ts`):
+
+```typescript
+import { setup } from '@cypress/puppeteer/support'
+```
+
+In your spec (e.g. `spec.cy.ts`):
+
+```typescript
+  it('switches to and tests a new tab', () => {
+    cy.visit('/')
+    cy.get('button').click() // opens a new tab
+
+    cy
+    .puppeteer('switchToTabAndGetContent')
+    .should('equal', 'You said: Hello from Page 1')
+  })
+```
+
+## API
+
+### Cypress Config - setup
+
+This sets up `@cypress/puppeteer` message handlers that run Puppeteer browser automation.
+
+```typescript
+setup(options)
+```
+
+#### Options
+
+- `on` _required_: The `on` event registration function provided by `setupNodeEvents`
+- `onMessage` _required_: An object with string keys and function values (see more details [below](#onmessage))
+- `puppeteer` _optional_: The `puppeteer` library imported from `puppeteer-core`, overriding the default version of `puppeteer-core` used by this plugin
+
+#### onMessage
```

Comment:
Do we want to explicitly call out that you can't run cypress commands in the message handlers?

Reply:
Yeah I like it. Addressed [here](https://github.com/cypress-io/cypress/pull/28370/commits/beb85942ff50c82e10405d142cc6b0bc820580c6#diff-bad8df9ac3ed2d72ad00ba7b71b5a9a6f51364a91eaf05c7ba1c9b1895fab279R113)